src/freertos_drivers/net_cc32xx/CC32xxSocket.cxx

```diff
@@ -1086,7 +1087,7 @@ int getaddrinfo(const char *nodename, const char *servname,
     std::unique_ptr<struct sockaddr> sa(new struct sockaddr);
     if (sa.get() == nullptr)
     {
-        free(*res);
+        ai.reset();
```

Comment:
this line is not necessary -- will happen automatically upon return. (The previous version was a bug though, good catch.)